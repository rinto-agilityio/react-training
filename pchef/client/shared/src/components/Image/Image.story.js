// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Image from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

storiesOf('Image', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Image
      url="https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg"
      handleTouch={action('clicked')}
    />
  ))
  .add('Disabled', () => (
    <Image
      url="https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg"
      disabled
    />
  ))
