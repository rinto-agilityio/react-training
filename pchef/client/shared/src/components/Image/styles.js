// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

const styles = StyleSheet.create({
  imageWrapper: {
    width: METRICS.mediumInput,
    height: METRICS.mediumInput,
    backgroundColor: '#ccc',
  },
  image: {
    width: '100%',
    height: '100%',
  },
})

export default styles
