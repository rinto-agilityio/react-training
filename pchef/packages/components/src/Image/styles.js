// Libs
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  image: {
    width: 40,
    height: 40
  },
});
