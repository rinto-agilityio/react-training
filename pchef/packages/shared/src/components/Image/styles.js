// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

const styles = StyleSheet.create({
  image: {
    width: METRICS.mediumInput,
    height: METRICS.mediumInput,
  },
})

export default styles