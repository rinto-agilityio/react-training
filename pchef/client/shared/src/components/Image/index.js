// Libs
import React, { useState, useEffect, memo, Suspense } from 'react'
import { Image as ImageNative, TouchableOpacity } from 'react-native'

// Styles
import styles from './styles'

// Components
import Loading from '../Loading'

type Props = {
  disabled?: boolean,
  handleTouch?: () => void,
  url: string,
  customBtnStyle?: {},
  customImageStyle?: {} | Array<{}>,
  resizeMethod?: string,
  thumbnail?: string
}

const placeholderImage = require('../../assets/images/placeholder.jpg')

const ImageComponent = ({
  url,
  thumbnail,
  disabled = false,
  customImageStyle,
  customBtnStyle = {},
  resizeMethod = 'auto',
  handleTouch = () => {},
}: Props) => {
  const [loadImg, setLoadImg] = useState({
    url: thumbnail || placeholderImage,
    loaded: false,
  })

  useEffect(() => {
    const imgLoader = new Image()

    imgLoader.onload = () => {
      setLoadImg({
        url: imgLoader.src,
        loaded: true,
      })
    }

    imgLoader.src = url
      ? url
      : placeholderImage

    return () => setLoadImg({ url, loaded: false })
  }, [url])

  return (
    <Suspense fallback={<Loading size="small" />}>
      <TouchableOpacity
        style={customBtnStyle}
        onPress={handleTouch}
        disabled={disabled}
      >
        <ImageNative
          source={{ uri: loadImg.url }}
          resizeMethod={resizeMethod}
          style={[styles.image, customImageStyle]}
        />
      </TouchableOpacity>
    </Suspense>
  )
}

export default memo(ImageComponent)
