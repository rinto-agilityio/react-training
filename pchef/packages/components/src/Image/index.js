// Libs
import React from 'react'
import { Image, TouchableOpacity } from 'react-native'

// Styles
import { styles } from './styles'

const ImageComponent = ({
  disabled,
  handleTouch,
  url,
  customBtnStyle,
  customImageStyle
}) => (
    <TouchableOpacity
      style={customBtnStyle}
      onPress={handleTouch}
      disabled={disabled}
    >
      <Image
        source={url}
        resizeMethod="resize"
        style={[styles.image, customImageStyle]}
      />
    </TouchableOpacity>
  )

ImageComponent.defaultProps = {
  disabled: false,
  customImageStyle: {},
  customBtnStyle: {},
  handleTouch: () => { },
  url: ''
}

export default ImageComponent