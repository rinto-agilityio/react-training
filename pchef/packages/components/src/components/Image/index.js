// Libs
import React from 'react'
import { Image, TouchableOpacity } from 'react-native'

// Styles
import { styles } from './styles'

const ImageComponent = ({
  disabled,
  handleTouch,
  url,
  customBtnStyle,
  customImageStyle,
  resizeMethod
}) => (
    <TouchableOpacity
      style={customBtnStyle}
      onPress={handleTouch}
      disabled={disabled}
    >
      <Image
        source={url}
        resizeMethod={resizeMethod}
        style={[styles.image, customImageStyle]}
      />
    </TouchableOpacity>
  )

ImageComponent.defaultProps = {
  disabled: false,
  customImageStyle: {},
  customBtnStyle: {},
  url: '',
  resizeMethod: 'auto',
  handleTouch: () => {}
}

export default ImageComponent