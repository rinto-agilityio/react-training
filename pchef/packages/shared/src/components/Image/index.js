// Libs
import React from 'react'
import { Image, TouchableOpacity } from 'react-native'

// Styles
import styles from './styles'

type Props = {
  disabled?: boolean,
  handleTouch?: () => void,
  url: string,
  customBtnStyle?: {},
  customImageStyle?: {} | Array<{}>,
  resizeMethod?: string,
}

const ImageComponent = ({
  disabled,
  handleTouch,
  url,
  customBtnStyle,
  customImageStyle,
  resizeMethod,
}: Props) => (
  <TouchableOpacity
    style={customBtnStyle}
    onPress={handleTouch}
    disabled={disabled}
  >
    <Image
      source={{ uri: url }}
      resizeMethod={resizeMethod}
      style={[styles.image, customImageStyle]}
    />
  </TouchableOpacity>
)

ImageComponent.defaultProps = {
  disabled: false,
  customImageStyle: {},
  customBtnStyle: {},
  resizeMethod: 'auto',
  handleTouch: () => {},
}

export default ImageComponent
