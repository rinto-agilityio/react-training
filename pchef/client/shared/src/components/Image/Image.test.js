// Libs
import renderer from 'react-test-renderer'

// Components
import Image from '.'

it('renders correctly', () => {
  const image = renderer.create(
    <Image
      url="https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg"
      disabled
    />,
  ).toJSON()

  expect(image).toMatchSnapshot()
});
