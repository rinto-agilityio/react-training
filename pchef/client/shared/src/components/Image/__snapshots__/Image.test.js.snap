// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  aria-disabled={true}
  className="css-view-1dbjc4n r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <div
    className="css-view-1dbjc4n r-backgroundColor-7klgxt r-height-eu3ka r-width-1aockid"
  >
    <div
      className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-1pi2tsx r-overflow-1udh08x r-width-13qz1uu r-zIndex-417010"
    >
      <div
        className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
        style={
          Object {
            "backgroundImage": "url(\\"https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg\\")",
          }
        }
      />
      <img
        alt=""
        className="css-accessibilityImage-9pa8cd"
        draggable={false}
        src="https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg"
      />
    </div>
  </div>
</div>
`;
